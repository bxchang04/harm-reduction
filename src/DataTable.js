import React from 'react';
import { useHistory } from 'react-router-dom';
import MaterialTable from 'material-table';
import { tableIcons } from './dataTableIcons';

// Head Row
const headStyle = {
  fontWeight: 'bold',
  fontSize: 'large',
};

// Body Rows
const rowStyle = {
  fontWeight: 'normal',
};

// detailPanel fields (each field)
const panelItemStyle = {
  paddingLeft: '6em', // to avoid tooltip
};

/**
 * DataTable (MUI MaterialTable wrapper)
 * @param {Array} columns: column definitions for table,
 * (see material-table docs for options), passed thru directly
 * except for 'detail' field. If .detail is true, data displays
 * in expandable DetailPanel only when activated.
 * @param {Array} rows: the data - direct pass-thru
 * @param {string} collectionName: the db collection, used for table
 * name and urls
 */
export default ({ columns, rows, collectionName }) => {
  const history = useHistory();
  const detailFields = columns.filter(col => col.detail === true);

  return (
    <MaterialTable
      title={`${collectionName}s`.toUpperCase()}
      icons={tableIcons}
      columns={columns}
      data={rows}
      options={{
        headerStyle: headStyle,
        rowStyle: rowStyle,
        pageSize: 5,
      }}
      actions={[
        {
          icon: tableIcons.View,
          tooltip: `View ${collectionName}`,
          onClick: (event, rowData) => {
            const url = `/${collectionName}s/${rowData.id}`;
            history.push(url);
          },
        },
      ]}
      detailPanel={
        detailFields.length > 0
          ? [
              {
                icon: tableIcons.Unfold,
                tooltip: 'Show Details',
                render: rowData =>
                  detailFields.map(
                    item =>
                      rowData[item.field] && (
                        <div style={panelItemStyle}>
                          {item.title}: {rowData[item.field]}
                        </div>
                      )
                  ),
              },
            ]
          : undefined
      }
    />
  );
};
