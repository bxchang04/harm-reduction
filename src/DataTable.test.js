import { render } from '@testing-library/react';
import React from 'react';
import { MemoryRouter } from 'react-router-dom';
import DataTable from './DataTable';

const columns = [
  { field: 'id', title: 'ID' },
  { field: 'name', title: 'Name' },
  { field: 'age', title: 'Age' },
];

const rows = [
  { id: 1, name: 'Dick', age: 10 },
  { id: 2, name: 'Jane', age: 12 },
  { id: 3, name: 'Spot', age: 2 },
];

test('Renders component', () => {
  const { baseElement } = render(
    <MemoryRouter>
      <DataTable columns={columns} rows={rows} collectionName="test" />
    </MemoryRouter>
  );
  expect(baseElement).toMatchSnapshot();
});
