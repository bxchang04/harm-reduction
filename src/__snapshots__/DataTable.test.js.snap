// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders component 1`] = `
<body>
  <div>
    <div
      class="MuiPaper-root MuiPaper-elevation2 MuiPaper-rounded"
      style="position: relative;"
    >
      <div
        class="MuiToolbar-root MuiToolbar-regular MTableToolbar-root-5 MuiToolbar-gutters"
      >
        <div
          class="MTableToolbar-title-9"
        >
          <h6
            class="MuiTypography-root MuiTypography-h6"
            style="white-space: nowrap; overflow: hidden; text-overflow: ellipsis;"
          >
            TESTS
          </h6>
        </div>
        <div
          class="MTableToolbar-spacer-7"
        />
        <div
          class="MuiFormControl-root MuiTextField-root MTableToolbar-searchField-10"
        >
          <div
            class="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-formControl MuiInput-formControl MuiInputBase-adornedStart MuiInputBase-adornedEnd"
          >
            <div
              class="MuiInputAdornment-root MuiInputAdornment-positionStart"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeSmall"
                focusable="false"
                title="Search"
                viewBox="0 0 24 24"
              >
                <path
                  d="M15.5 14h-.79l-.28-.27C15.41 12.59 16 11.11 16 9.5 16 5.91 13.09 3 9.5 3S3 5.91 3 9.5 5.91 16 9.5 16c1.61 0 3.09-.59 4.23-1.57l.27.28v.79l5 4.99L20.49 19l-4.99-5zm-6 0C7.01 14 5 11.99 5 9.5S7.01 5 9.5 5 14 7.01 14 9.5 11.99 14 9.5 14z"
                />
              </svg>
            </div>
            <input
              aria-invalid="false"
              aria-label="Search"
              class="MuiInputBase-input MuiInput-input MuiInputBase-inputAdornedStart MuiInputBase-inputAdornedEnd"
              placeholder="Search"
              type="text"
              value=""
            />
            <div
              class="MuiInputAdornment-root MuiInputAdornment-positionEnd"
            >
              <button
                class="MuiButtonBase-root MuiIconButton-root Mui-disabled Mui-disabled"
                disabled=""
                tabindex="-1"
                type="button"
              >
                <span
                  class="MuiIconButton-label"
                >
                  <svg
                    aria-hidden="true"
                    class="MuiSvgIcon-root MuiSvgIcon-fontSizeSmall"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M19 6.41L17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12z"
                    />
                  </svg>
                </span>
              </button>
            </div>
          </div>
        </div>
        <div
          class="MTableToolbar-actions-8"
        >
          <div>
            <div>
              <span />
            </div>
          </div>
        </div>
      </div>
      <div
        class="Component-horizontalScrollContainer-12"
        style="overflow-x: auto; position: relative;"
      >
        <div>
          <div
            style="overflow-y: auto;"
          >
            <div>
              <table
                class="MuiTable-root"
                style="table-layout: auto;"
              >
                <thead
                  class="MuiTableHead-root"
                >
                  <tr
                    class="MuiTableRow-root MuiTableRow-head"
                  >
                    <th
                      class="MuiTableCell-root MuiTableCell-head MTableHeader-header-13 MuiTableCell-paddingCheckbox"
                      scope="col"
                      style="font-weight: bold; font-size: large; width: 48px; text-align: center; box-sizing: border-box;"
                    >
                      <span
                        aria-disabled="true"
                        class="MuiButtonBase-root MuiTableSortLabel-root Mui-disabled"
                        role="button"
                        tabindex="-1"
                      >
                        Actions
                      </span>
                    </th>
                    <th
                      class="MuiTableCell-root MuiTableCell-head MTableHeader-header-13 MuiTableCell-alignLeft"
                      scope="col"
                      style="font-weight: bold; font-size: large; box-sizing: border-box;"
                    >
                      <span
                        aria-disabled="false"
                        class="MuiButtonBase-root MuiTableSortLabel-root"
                        role="button"
                        tabindex="0"
                      >
                        <div
                          aria-describedby="rbd-hidden-text-0-hidden-text-0"
                          data-rbd-drag-handle-context-id="0"
                          data-rbd-drag-handle-draggable-id="0"
                          data-rbd-draggable-context-id="0"
                          data-rbd-draggable-id="0"
                          draggable="false"
                          role="button"
                          tabindex="0"
                        >
                          ID
                        </div>
                        <svg
                          aria-hidden="true"
                          class="MuiSvgIcon-root MuiTableSortLabel-icon MuiTableSortLabel-iconDirectionAsc"
                          focusable="false"
                          viewBox="0 0 24 24"
                        >
                          <path
                            d="M20 12l-1.41-1.41L13 16.17V4h-2v12.17l-5.58-5.59L4 12l8 8 8-8z"
                          />
                        </svg>
                      </span>
                    </th>
                    <th
                      class="MuiTableCell-root MuiTableCell-head MTableHeader-header-13 MuiTableCell-alignLeft"
                      scope="col"
                      style="font-weight: bold; font-size: large; box-sizing: border-box;"
                    >
                      <span
                        aria-disabled="false"
                        class="MuiButtonBase-root MuiTableSortLabel-root"
                        role="button"
                        tabindex="0"
                      >
                        <div
                          aria-describedby="rbd-hidden-text-0-hidden-text-0"
                          data-rbd-drag-handle-context-id="0"
                          data-rbd-drag-handle-draggable-id="1"
                          data-rbd-draggable-context-id="0"
                          data-rbd-draggable-id="1"
                          draggable="false"
                          role="button"
                          tabindex="0"
                        >
                          Name
                        </div>
                        <svg
                          aria-hidden="true"
                          class="MuiSvgIcon-root MuiTableSortLabel-icon MuiTableSortLabel-iconDirectionAsc"
                          focusable="false"
                          viewBox="0 0 24 24"
                        >
                          <path
                            d="M20 12l-1.41-1.41L13 16.17V4h-2v12.17l-5.58-5.59L4 12l8 8 8-8z"
                          />
                        </svg>
                      </span>
                    </th>
                    <th
                      class="MuiTableCell-root MuiTableCell-head MTableHeader-header-13 MuiTableCell-alignLeft"
                      scope="col"
                      style="font-weight: bold; font-size: large; box-sizing: border-box;"
                    >
                      <span
                        aria-disabled="false"
                        class="MuiButtonBase-root MuiTableSortLabel-root"
                        role="button"
                        tabindex="0"
                      >
                        <div
                          aria-describedby="rbd-hidden-text-0-hidden-text-0"
                          data-rbd-drag-handle-context-id="0"
                          data-rbd-drag-handle-draggable-id="2"
                          data-rbd-draggable-context-id="0"
                          data-rbd-draggable-id="2"
                          draggable="false"
                          role="button"
                          tabindex="0"
                        >
                          Age
                        </div>
                        <svg
                          aria-hidden="true"
                          class="MuiSvgIcon-root MuiTableSortLabel-icon MuiTableSortLabel-iconDirectionAsc"
                          focusable="false"
                          viewBox="0 0 24 24"
                        >
                          <path
                            d="M20 12l-1.41-1.41L13 16.17V4h-2v12.17l-5.58-5.59L4 12l8 8 8-8z"
                          />
                        </svg>
                      </span>
                    </th>
                  </tr>
                </thead>
                <tbody
                  class="MuiTableBody-root"
                >
                  <tr
                    class="MuiTableRow-root"
                    index="0"
                    level="0"
                    path="0"
                    style="transition: all ease 300ms; font-weight: normal;"
                  >
                    <td
                      class="MuiTableCell-root MuiTableCell-body MuiTableCell-paddingNone"
                      style="width: 48px; padding: 0px 5px; box-sizing: border-box;"
                    >
                      <div
                        style="display: flex;"
                      >
                        <button
                          class="MuiButtonBase-root MuiIconButton-root MuiIconButton-colorInherit"
                          tabindex="0"
                          title="View test"
                          type="button"
                        >
                          <span
                            class="MuiIconButton-label"
                          >
                            <svg
                              aria-hidden="true"
                              class="MuiSvgIcon-root"
                              focusable="false"
                              viewBox="0 0 24 24"
                            >
                              <path
                                d="M11.5 9C10.12 9 9 10.12 9 11.5s1.12 2.5 2.5 2.5 2.5-1.12 2.5-2.5S12.88 9 11.5 9zM20 4H4c-1.1 0-2 .9-2 2v12c0 1.1.9 2 2 2h16c1.1 0 2-.9 2-2V6c0-1.1-.9-2-2-2zm-3.21 14.21l-2.91-2.91c-.69.44-1.51.7-2.39.7C9.01 16 7 13.99 7 11.5S9.01 7 11.5 7 16 9.01 16 11.5c0 .88-.26 1.69-.7 2.39l2.91 2.9-1.42 1.42z"
                              />
                            </svg>
                          </span>
                          <span
                            class="MuiTouchRipple-root"
                          />
                        </button>
                      </div>
                    </td>
                    <td
                      class="MuiTableCell-root MuiTableCell-body MuiTableCell-alignLeft"
                      style="box-sizing: border-box; font-size: inherit; font-family: inherit; font-weight: inherit;"
                      value="1"
                    >
                      1
                    </td>
                    <td
                      class="MuiTableCell-root MuiTableCell-body MuiTableCell-alignLeft"
                      style="box-sizing: border-box; font-size: inherit; font-family: inherit; font-weight: inherit;"
                      value="Dick"
                    >
                      Dick
                    </td>
                    <td
                      class="MuiTableCell-root MuiTableCell-body MuiTableCell-alignLeft"
                      style="box-sizing: border-box; font-size: inherit; font-family: inherit; font-weight: inherit;"
                      value="10"
                    >
                      10
                    </td>
                  </tr>
                  <tr
                    class="MuiTableRow-root"
                    index="1"
                    level="0"
                    path="1"
                    style="transition: all ease 300ms; font-weight: normal;"
                  >
                    <td
                      class="MuiTableCell-root MuiTableCell-body MuiTableCell-paddingNone"
                      style="width: 48px; padding: 0px 5px; box-sizing: border-box;"
                    >
                      <div
                        style="display: flex;"
                      >
                        <button
                          class="MuiButtonBase-root MuiIconButton-root MuiIconButton-colorInherit"
                          tabindex="0"
                          title="View test"
                          type="button"
                        >
                          <span
                            class="MuiIconButton-label"
                          >
                            <svg
                              aria-hidden="true"
                              class="MuiSvgIcon-root"
                              focusable="false"
                              viewBox="0 0 24 24"
                            >
                              <path
                                d="M11.5 9C10.12 9 9 10.12 9 11.5s1.12 2.5 2.5 2.5 2.5-1.12 2.5-2.5S12.88 9 11.5 9zM20 4H4c-1.1 0-2 .9-2 2v12c0 1.1.9 2 2 2h16c1.1 0 2-.9 2-2V6c0-1.1-.9-2-2-2zm-3.21 14.21l-2.91-2.91c-.69.44-1.51.7-2.39.7C9.01 16 7 13.99 7 11.5S9.01 7 11.5 7 16 9.01 16 11.5c0 .88-.26 1.69-.7 2.39l2.91 2.9-1.42 1.42z"
                              />
                            </svg>
                          </span>
                          <span
                            class="MuiTouchRipple-root"
                          />
                        </button>
                      </div>
                    </td>
                    <td
                      class="MuiTableCell-root MuiTableCell-body MuiTableCell-alignLeft"
                      style="box-sizing: border-box; font-size: inherit; font-family: inherit; font-weight: inherit;"
                      value="2"
                    >
                      2
                    </td>
                    <td
                      class="MuiTableCell-root MuiTableCell-body MuiTableCell-alignLeft"
                      style="box-sizing: border-box; font-size: inherit; font-family: inherit; font-weight: inherit;"
                      value="Jane"
                    >
                      Jane
                    </td>
                    <td
                      class="MuiTableCell-root MuiTableCell-body MuiTableCell-alignLeft"
                      style="box-sizing: border-box; font-size: inherit; font-family: inherit; font-weight: inherit;"
                      value="12"
                    >
                      12
                    </td>
                  </tr>
                  <tr
                    class="MuiTableRow-root"
                    index="2"
                    level="0"
                    path="2"
                    style="transition: all ease 300ms; font-weight: normal;"
                  >
                    <td
                      class="MuiTableCell-root MuiTableCell-body MuiTableCell-paddingNone"
                      style="width: 48px; padding: 0px 5px; box-sizing: border-box;"
                    >
                      <div
                        style="display: flex;"
                      >
                        <button
                          class="MuiButtonBase-root MuiIconButton-root MuiIconButton-colorInherit"
                          tabindex="0"
                          title="View test"
                          type="button"
                        >
                          <span
                            class="MuiIconButton-label"
                          >
                            <svg
                              aria-hidden="true"
                              class="MuiSvgIcon-root"
                              focusable="false"
                              viewBox="0 0 24 24"
                            >
                              <path
                                d="M11.5 9C10.12 9 9 10.12 9 11.5s1.12 2.5 2.5 2.5 2.5-1.12 2.5-2.5S12.88 9 11.5 9zM20 4H4c-1.1 0-2 .9-2 2v12c0 1.1.9 2 2 2h16c1.1 0 2-.9 2-2V6c0-1.1-.9-2-2-2zm-3.21 14.21l-2.91-2.91c-.69.44-1.51.7-2.39.7C9.01 16 7 13.99 7 11.5S9.01 7 11.5 7 16 9.01 16 11.5c0 .88-.26 1.69-.7 2.39l2.91 2.9-1.42 1.42z"
                              />
                            </svg>
                          </span>
                          <span
                            class="MuiTouchRipple-root"
                          />
                        </button>
                      </div>
                    </td>
                    <td
                      class="MuiTableCell-root MuiTableCell-body MuiTableCell-alignLeft"
                      style="box-sizing: border-box; font-size: inherit; font-family: inherit; font-weight: inherit;"
                      value="3"
                    >
                      3
                    </td>
                    <td
                      class="MuiTableCell-root MuiTableCell-body MuiTableCell-alignLeft"
                      style="box-sizing: border-box; font-size: inherit; font-family: inherit; font-weight: inherit;"
                      value="Spot"
                    >
                      Spot
                    </td>
                    <td
                      class="MuiTableCell-root MuiTableCell-body MuiTableCell-alignLeft"
                      style="box-sizing: border-box; font-size: inherit; font-family: inherit; font-weight: inherit;"
                      value="2"
                    >
                      2
                    </td>
                  </tr>
                  <tr
                    class="MuiTableRow-root"
                    style="height: 49px;"
                  />
                  <tr
                    class="MuiTableRow-root"
                    style="height: 49px;"
                  />
                  <tr
                    class="MuiTableRow-root"
                    style="height: 1px;"
                  />
                </tbody>
              </table>
            </div>
          </div>
        </div>
      </div>
      <table
        class="MuiTable-root"
      >
        <tfoot
          class="MuiTableFooter-root"
          style="display: grid;"
        >
          <tr
            class="MuiTableRow-root MuiTableRow-footer"
          >
            <td
              class="MuiTableCell-root MuiTableCell-footer MuiTablePagination-root Component-paginationRoot-1"
              colspan="3"
              icons="[object Object]"
              style="float: right; overflow-x: auto;"
            >
              <div
                class="MuiToolbar-root MuiToolbar-regular MuiTablePagination-toolbar Component-paginationToolbar-2 MuiToolbar-gutters"
              >
                <div
                  class="MuiTablePagination-spacer"
                />
                <p
                  class="MuiTypography-root MuiTablePagination-caption Component-paginationCaption-3 MuiTypography-body2 MuiTypography-colorInherit"
                >
                  Rows per page:
                </p>
                <div
                  class="MuiInputBase-root MuiTablePagination-input MuiTablePagination-selectRoot Component-paginationSelectRoot-4"
                >
                  <div
                    aria-haspopup="listbox"
                    aria-label="Rows per page:"
                    class="MuiSelect-root MuiSelect-select MuiTablePagination-select MuiSelect-selectMenu MuiInputBase-input"
                    role="button"
                    tabindex="0"
                  >
                    <div
                      style="padding: 0px 5px;"
                    >
                      5 rows 
                    </div>
                  </div>
                  <input
                    type="hidden"
                    value="5"
                  />
                  <svg
                    aria-hidden="true"
                    class="MuiSvgIcon-root MuiSelect-icon MuiTablePagination-selectIcon"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M7 10l5 5 5-5z"
                    />
                  </svg>
                </div>
                <p
                  class="MuiTypography-root MuiTablePagination-caption Component-paginationCaption-3 MuiTypography-body2 MuiTypography-colorInherit"
                >
                  1-3 of 3
                </p>
                <div
                  class="MTablePaginationInner-root-14"
                >
                  <span
                    class=""
                    title="First Page"
                  >
                    <button
                      class="MuiButtonBase-root MuiIconButton-root Mui-disabled Mui-disabled"
                      disabled=""
                      tabindex="-1"
                      type="button"
                    >
                      <span
                        class="MuiIconButton-label"
                      >
                        <svg
                          aria-hidden="true"
                          class="MuiSvgIcon-root"
                          focusable="false"
                          viewBox="0 0 24 24"
                        >
                          <path
                            d="M18.41 16.59L13.82 12l4.59-4.59L17 6l-6 6 6 6zM6 6h2v12H6z"
                          />
                        </svg>
                      </span>
                    </button>
                  </span>
                  <span
                    class=""
                    title="Previous Page"
                  >
                    <button
                      class="MuiButtonBase-root MuiIconButton-root Mui-disabled Mui-disabled"
                      disabled=""
                      tabindex="-1"
                      type="button"
                    >
                      <span
                        class="MuiIconButton-label"
                      >
                        <svg
                          aria-hidden="true"
                          class="MuiSvgIcon-root"
                          focusable="false"
                          viewBox="0 0 24 24"
                        >
                          <path
                            d="M15.41 7.41L14 6l-6 6 6 6 1.41-1.41L10.83 12z"
                          />
                        </svg>
                      </span>
                    </button>
                  </span>
                  <span
                    class="MuiTypography-root MuiTypography-caption"
                    style="flex: 1; text-align: center; align-self: center; flex-basis: inherit;"
                  >
                    1-3 of 3
                  </span>
                  <span
                    class=""
                    title="Next Page"
                  >
                    <button
                      class="MuiButtonBase-root MuiIconButton-root Mui-disabled Mui-disabled"
                      disabled=""
                      tabindex="-1"
                      type="button"
                    >
                      <span
                        class="MuiIconButton-label"
                      >
                        <svg
                          aria-hidden="true"
                          class="MuiSvgIcon-root"
                          focusable="false"
                          viewBox="0 0 24 24"
                        >
                          <path
                            d="M10 6L8.59 7.41 13.17 12l-4.58 4.59L10 18l6-6z"
                          />
                        </svg>
                      </span>
                    </button>
                  </span>
                  <span
                    class=""
                    title="Last Page"
                  >
                    <button
                      class="MuiButtonBase-root MuiIconButton-root Mui-disabled Mui-disabled"
                      disabled=""
                      tabindex="-1"
                      type="button"
                    >
                      <span
                        class="MuiIconButton-label"
                      >
                        <svg
                          aria-hidden="true"
                          class="MuiSvgIcon-root"
                          focusable="false"
                          viewBox="0 0 24 24"
                        >
                          <path
                            d="M5.59 7.41L10.18 12l-4.59 4.59L7 18l6-6-6-6zM16 6h2v12h-2z"
                          />
                        </svg>
                      </span>
                    </button>
                  </span>
                </div>
              </div>
            </td>
          </tr>
        </tfoot>
      </table>
    </div>
  </div>
  <div
    aria-atomic="true"
    aria-live="assertive"
    id="rbd-announcement-0"
    style="position: absolute; width: 1px; height: 1px; margin: -1px; border: 0px; padding: 0px; overflow: hidden; clip-path: inset(100%);"
  />
  <div
    id="rbd-hidden-text-0-hidden-text-0"
    style="display: none;"
  >
    
  Press space bar to start a drag.
  When dragging you can use the arrow keys to move the item around and escape to cancel.
  Some screen readers may require you to be in focus mode or to use your pass through key

  </div>
</body>
`;
